### PR TITLE
docs: add DatoCMS recipe

### PR DESCRIPTION
**Pull request by jdiogopeixoto**
This PR adds a recipe on how to setup and integrate DatoCMS, covering important steps that might be missed along the way.

### CODE REVIEW

**Comment by acostalima** [comment]
When your branch is still WIP, please create a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) instead. Thanks!

File: docusaurus/docs/recipes/dato-cms.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+---
+
+Often there is a need to have a **Content Management System (CMS)** to manage all the website content in a easy way for non tech people.
+In this recipe you will be guided on how to setup and integrate **DatoCMS**.
+
+## Walk-through
+
+### 1. Content modeling
+
+This is an important part of this recipe. All the models should be user friendly as they are often used by people with few technical knowledge. In **DatoCMS**, after creating a project, you can create new models in "Settings" > "Models".
```
Comment (rafaelramalho19) [inline docusaurus/docs/recipes/dato-cms.md:14]: "with less technical knowledge"

**Comment by satazor** [comment]
Great job, looks much better!

File: docusaurus/static/recipes-assets/dato-cms/with-apollo/withApollo.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    // We send the Apollo Client as a prop to the component to avoid calling initApollo() twice in the server.
+    // Otherwise, the component would have to call initApollo() again but this
+    // time without the context. Once that happens, the following code will make sure we send
+    // the prop as `null` to the browser.
+    apolloClient.toJSON = () => null;
+
+    // Add apolloClient to NextPageContext & NextAppContext.
+    // This allows us to consume the apolloClient inside our
+    // custom `getInitialProps({ apolloClient })`.
+    ctx.apolloClient = apolloClient;
```
Comment (satazor) [inline docusaurus/static/recipes-assets/dato-cms/with-apollo/withApollo.js:40]:
```suggestion

    ctx.apolloClient = apolloClient;
```

File: docusaurus/docs/recipes/dato-cms.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+When creating a new field type for your models you should:
+- Choose a descriptive and user friendly name;
+- Select the option "Enable localization on this field?" in case you want it to be translatable;
+- Add all the necessary validations - e.g. mark the field as required or unique, limit the number of records;
+- Add an "Help text" in the "Presentation" tab giving more information about the field.
+
+### 2. How to include DatoCMS in your App
+
+First you need to get your **API token**, you can find it in **DatoCMS** under "Settings" > "API tokens". Then you need to create a new environment variable called `DATOCMS_TOKEN` with the token - add it to `.env` and `next.config.js` files.
```
[the remark this answers is not in the document]
Comment (jdiogopeixoto) [reply docusaurus/docs/recipes/dato-cms.md]: Added a reference to our docs.

File: docusaurus/docs/recipes/dato-cms.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+
+After that you need a **GraphQL client** to access the Content Delivery API - `https://graphql.datocms.com`.
+For this recipe we will use **Apollo** but feel free to use other alternatives.
+
+You will need the following packages:
+```sh
+npm i apollo-client apollo-cache-inmemory apollo-link-http @apollo/react-hooks @apollo/react-ssr isomorphic-unfetch graphql graphql-tag
```
[the remark this answers is not in the document]
Comment (jdiogopeixoto) [reply docusaurus/docs/recipes/dato-cms.md:41]: The package is not necessary but the `cache` property is necessary in the Apollo Client. Although it is possible to change the policy: https://www.apollographql.com/docs/react/api/react-apollo/#optionsfetchpolicy. 

Should we set it to `network-only`?

**Pull request merged**